### PR TITLE
Fix #448 switch to pyproject.toml

### LINKED ISSUE #448: Convert setup.py to pyproject.toml

### LINKED ISSUE #453: Check for pyproject.toml or setup.py
pkcli.cli only checks for setup.py

### PR DESCRIPTION
**Pull request by robnagler**
- Fix #453 pkcli.ci finds pyproject.toml or setup.py
- Partial #451 emptied pytest_plugin
- Converted README.md (from rst)

### CODE REVIEW

**Review by e-carlin — CHANGES_REQUESTED** [review]
I tried creating/installing a new package and got the error below
```
[py3;@e5f2fce0e61a radiasoft]$ mkdir foo
[py3;@e5f2fce0e61a radiasoft]$ cd foo
[py3;@e5f2fce0e61a foo]$ git init
Initialized empty Git repository in /home/vagrant/src/radiasoft/foo/.git/
[py3;@e5f2fce0e61a foo]$ pykern projex init-rs-tree 'egc description'
pykern projex init-rs-tree 'egc description'
[py3;@e5f2fce0e61a foo]$ pip install -e .
pip install -e .
Obtaining file:///home/vagrant/src/radiasoft/foo
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error

  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [30 lines of output]
      fatal: ambiguous argument 'HEAD': unknown revision or path not in the working tree.
      Use '--' to separate paths from revisions, like this:
      'git <command> [<revision>...] -- [<file>...]'
      Traceback (most recent call last):
        File "/home/vagrant/src/radiasoft/pykern/pykern/pksetup.py", line 414, in _check_output
          res = subprocess.check_output(*args, **kwargs)
        File "/home/vagrant/.pyenv/versions/3.9.15/lib/python3.9/subprocess.py", line 424, in check_output
          return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
        File "/home/vagrant/.pyenv/versions/3.9.15/lib/python3.9/subprocess.py", line 528, in run
          raise CalledProcessError(retcode, process.args,
      subprocess.CalledProcessError: Command '['git', 'rev-parse', '--abbrev-ref', 'HEAD']' returned non-zero exit status 128.

      During handling of the above exception, another exception occurred:

      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "/home/vagrant/src/radiasoft/foo/setup.py", line 9, in <module>
          pykern.pksetup.setup(
        File "/home/vagrant/src/radiasoft/pykern/pykern/pksetup.py", line 390, in setup
          base = _state(base, kwargs)
        File "/home/vagrant/src/radiasoft/pykern/pykern/pksetup.py", line 653, in _state
          state["version"], s = _version(base)
        File "/home/vagrant/src/radiasoft/pykern/pykern/pksetup.py", line 703, in _version
          v2, sha = _version_from_git(base)
        File "/home/vagrant/src/radiasoft/pykern/pykern/pksetup.py", line 750, in _version_from_git
          branch = _check_output(["git", "rev-parse", "--abbrev-ref", "HEAD"]).rstrip()
        File "/home/vagrant/src/radiasoft/pykern/pykern/pksetup.py", line 420, in _check_output
          sys.stderr.write(e.output)
      TypeError: write() argument must be str, not bytes
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip
```

**Comment by e-carlin** [comment]
Disregard my previous message. I needed to have a commit for the subprocess to work.

File: pykern/pkunit.py [hunk 1]
```diff
@@ -1,10 +1,9 @@
-# -*- coding: utf-8 -*-
 """Useful operations for unit tests
 
 :copyright: Copyright (c) 2015 RadiaSoft LLC.  All Rights Reserved.
 :license: http://www.apache.org/licenses/LICENSE-2.0.html
 """
-from __future__ import absolute_import, division, print_function
+
```
Comment (e-carlin) [inline pykern/pkunit.py:6]: Remove

File: pykern/pkcollections.py [hunk 2]
```diff
@@ -50,11 +50,11 @@ class PKDict(dict):
     def __copy__(self):
         return self.__class__(self)
 
-    def __deepcopy__(self, memo):
-        rv = self.copy()
-        for k, v in rv.items():
-            rv[copy.deepcopy(k, memo)] = copy.deepcopy(v, memo)
-        return rv
+    #    def __deepcopy__(self, memo):
```
Comment (e-carlin) [inline pykern/pkcollections.py:53]: eliminate?

**Review by e-carlin — APPROVED** [review]
Two nits

File: pykern/pkunit.py (hunk 1 above)
Comment (robnagler) [reply pykern/pkunit.py:6]: Remove what?

Comment (e-carlin) [reply pykern/pkunit.py:6]: The blank space between the doc string and imports

Comment (robnagler) [reply pykern/pkunit.py:6]: That's a new feature of black.

**Pull request merged**